packages/dev/babel-preset/index.js

```diff
@@ -1,36 +1,63 @@
-module.exports = () => ({
-  presets: [
-    [
-      require('@babel/preset-env'),
-      {
-        modules: false,
-        targets: {
-          node: 12,
-        },
-      },
-    ],
-    require('@babel/preset-react'),
-    require('@babel/preset-flow'),
-  ],
-  plugins: [
-    [
-      require('@babel/plugin-transform-modules-commonjs'),
-      {
-        lazy: true,
-      },
-    ],
-  ],
-  env: {
-    production: {
+module.exports = api => {
+  let name = api.caller(caller => caller && caller.name);
+  if (name === 'parcel') {
```

Comment:
A different Babel config is used when Parcel is building itself since we don't want to compile modules etc.